src/core/server/opensearch/version_check/ensure_opensearch_version.test.ts

```diff
@@ -346,6 +351,149 @@ describe('pollOpenSearchNodesVersion', () => {
       .pipe(take(1))
       .subscribe({
         next: (result) => {
+          expect(internalClient.nodes.info).toBeCalledWith({
+            node_id: '_local',
+            metric: 'process',
+            filter_path: ['nodes.*.version', 'nodes.*.http.publish_address', 'nodes.*.ip'],
+          });
+          expect(result).toEqual(
+            mapNodesVersionCompatibility(filteredNodes, OPENSEARCH_DASHBOARDS_VERSION, false)
+          );
+          expect(result.isCompatible).toBe(false);
+        },
+        complete: done,
+        error: done,
+      });
+  });
+
+  it('returns compatibility results and isCompatible=true with filters and multiple cluster ids', (done) => {
```

Comment:
Hey, I don't see much differences for this test and the one below it. Both of them check `expect(result.isCompatible).toBe(false);`. The only difference is the version for `nodes[node-1]`

```
nodes['node-1'] = {
      version: '5.1.0',
      http: {
        publish_address: 'http_address',
      },
      ip: 'ip',
      attributes: {
        cluster_id: '1',
        custom_attribute: 'foo',
      },
    };

```
and below one 
```
nodes['node-1'] = {
      version: '6.2.0',
      http: {
        publish_address: 'http_address',
      },
      ip: 'ip',
      attributes: {
        cluster_id: '1',
        custom_attribute: 'foo',
      },
    };
```

So my question would be why we have two similar tests on `isCompatible=false`?

Reply:
good catch the first should have been true. got too focused on what was calling the client info to verify the result. fixed.

both tests filter out `node-2`. the first one should have been true because `node-0` and `node-1` have different cluster ids but the same version. So that is valid for OpenSearch Dashboards to keep running. But the second is false because `node-0` and `node-1` have different cluster ids and different versions so OpenSearch Dashboards should go down.